### DISCUSSION TITLE
PolyGlot: Language Construction Toolkit 3.5

### DISCUSSION THREAD

**Discussion by DraqueT** [DraqueT/PolyGlot#1148] [General]
Minor release here that cleans up a few things from the last big release and adds a few features I had wanted to include with 3.5 in the first place.

Enjoy, everyone!

FEATURES ADDED
- Java 8 no longer needed for Excel features and PDF generation
- Revised invalid entry display in some menus
- Added ability to scale UI in Windows and Linux
- Word generator now compatible with comma delimited lists to allow for multi symbol characters
- Word generator now has visual feedback to help with input formatting
- Word generator now saves settings rather than discarding on close of window

BUGS FIXED
- Tooltips were severely broken on Windows/Linux, not functioning at all
- Under certain circumstances, add part of speech button could become temporarily disabled
- UI scaling in HD environments was broken
- Part of Speech mandatory can no longer be set if none are defined
- Import menu was not static
- Error message occasionally/popped on startup for null ConWord value

<hr /><em>This discussion was created from the release <a href='https://github.com/DraqueT/PolyGlot/releases/tag/'>PolyGlot: Language Construction Toolkit 3.5</a>.</em>

**Comment by AnonAmy** [comment]
When I use the "Publish Language To PDF" feature with this new version, the pop up dialogue page can't be resized, does not fit my screen, and so I am unable to move my mouse down to click a print button (which I can't see since its past the bottom of my screen). 

Also is there a word count feature, which tells us how many words we have in our lexicon?

This is an awesome program!

**Reply by TrapinchO** [reply]
Please create a new issue.
Regarding the word count, it is in language report, though I don't think it is in the app. Feel free to create a new issue about it.

**Comment by SpartanCreeper** [comment]
hello, is it possible to reduplicate part of the text as a conjugation?
thx for making this btw, its amazing

**Reply by SpartanCreeper** [reply]
and another question: is it possible for conjugation to come between vowels, cause i got the conjugation -m to mark a noun to be of the opposite sex, but if the words end in 2 vowels it becomes VmV, can i create this in the autogeneration?

**Reply by SpartanCreeper** [reply]
maybe change the regex entirely? lexurgy has an amazing system in which they have less symbols and easier over all, they work with classes etc but that isnt necessary, u could do something like instead of "$" u have "\_ $", in which "\_" resembles the place and "$" the word boundry, so "$ \_" is the same as "^", u could add things like regrex: "\_ [a, i, u]? $"; replacement: "m" which is similar to the regex, but far easier (tho i still dont know how to to do that with the regrex)
lexurgy seems to be a simplified version of the regex u r using, so itd be amazing if ud add that
how lexurgy works: https://www.meamoria.com/lexurgy/html/sc-tutorial.html (it is a sound change generator, but a big part could be used here)

**Reply by SpartanCreeper** [reply]
also by the word generator instead of having for example C=mnptk u have C = m / n / p / t / k
the spaces wouldnt count, and if u want a space u do $$ like C = m / n / $$ / p / t / k
"$$" comes from lexurgy btw

**Reply by TrapinchO** [reply]
Perhaps it would be better to open an issue for the suggestions, where it is possible to better keep track of them.

> also by the word generator instead of having for example C=mnptk u have C = m / n / p / t / k
> the spaces wouldnt count, and if u want a space u do $$ like C = m / n / $$ / p / t / k
> "$$" comes from lexurgy btw

This has been already suggested and is planned for inclusion.

**Reply by SpartanCreeper** [reply]
ok thx, ill put it in there

**Reply by SpartanCreeper** [reply]
and about the reduplication and conjugation within the word, is that possible already?

**Reply by TrapinchO** [reply]
> and about the reduplication and conjugation within the word, is that possible already?

Almost surely. Check the examples provided within the program, iirc there might be showcase of this.

**Reply by SpartanCreeper** [reply]
where? i cant find them

**Reply by TrapinchO** [reply]
> where? i cant find them

Toolbar -> Help -> Example Languages

**Reply by SpartanCreeper** [reply]
thx

**Reply by SpartanCreeper** [reply]
![image](https://user-images.githubusercontent.com/65303062/170876527-5457fea2-7f56-4d91-8f10-7e812e916500.png)
tf is this lmao what does it mean

**Reply by SpartanCreeper** [reply]
![image](https://user-images.githubusercontent.com/65303062/170876632-9f456aea-0b4b-4613-9672-729021596345.png)
and whats this for magic, is it so hard to make infixes?
